config/routes.rb

```diff
@@ -304,10 +297,91 @@
   get "signup" => "static#show", id: "signup", constraints: { format: /(json|html)/ }
   get "login-preferences" => "static#show", id: "login", constraints: { format: /(json|html)/ }
 
+  get "my/*path", to: 'users#my_redirect'
+  get "user_preferences" => "users#user_preferences_redirect"
+
+  # New /u/ routes
```

Comment:
Maybe something like that? https://github.com/tgxworld/discourse/commit/d04f6c2ff7a4d11378ddcd2e29803aea259badcd